tests/integration/relationships/has-many-test.js

```diff
@@ -1528,6 +1528,25 @@ test("Polymorphic relationships with a hasMany is set up correctly on both sides
   assert.equal(get(email, 'posts.length'), 1, "The inverse has many is set up correctly on the email side.");
 });
 
+test("Can push a polymorphic belongsTo, the relationship is set up correctly on both sides", function(assert) {
+  assert.expect(2);
+
+  Comment.reopen({
+    message: DS.belongsTo('message', { polymorphic: true, async: false })
+  });
+  var post, comment;
+
+  run(function () {
+    post = env.store.push({ data: { type: 'post', id: 1 } });
+    comment = env.store.push({ data: { type: 'comment', id: 1 } });
+    let comments = post.get('comments');
+    comments.pushObject(comment);
+  });
+
+  assert.equal(comment.get('message'), post, 'The polymorphic belongsTo is set up correctly');
```

Comment:
Unfortunately, `assert.equal` uses an equality algorithm that totally fails on computed properties. The tests fare failing because `assert.equal` is being called with a model which ultimately has `store` injected on it which has `defaultAdapter` as a computed property. I would refactor this test to one capable of a strict reference equality check.

Reply:
Thanks for the explanation, I will use test the equality on the id instead.